nifi-nar-bundles/nifi-marklogic-bundle/nifi-marklogic-processors/src/test/java/org/apache/nifi/marklogic/processor/QueryMarkLogicTest.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.marklogic.processor;
+
+import org.apache.nifi.reporting.InitializationException;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.marklogic.client.io.Format;
+import com.marklogic.client.io.StringHandle;
+import com.marklogic.client.query.RawCombinedQueryDefinition;
+import com.marklogic.client.query.RawStructuredQueryDefinition;
+import com.marklogic.client.query.StructuredQueryDefinition;
+
+public class QueryMarkLogicTest extends AbstractMarkLogicProcessorTest {
+
+    private TestQueryMarkLogic processor;
+
+    @Before
+    public void setup() throws InitializationException {
+        processor = new TestQueryMarkLogic();
+        initialize(processor);
+        processContext.setProperty(TestQueryMarkLogic.THREAD_COUNT.getName(), "5");
+        processContext.setProperty(TestQueryMarkLogic.BATCH_SIZE, "15");
+        processContext.setProperty(TestQueryMarkLogic.QUERY_TYPE, QueryMarkLogic.QueryTypes.COLLECTION.name());
```

Comment:
Each of the tests except for `testCollectionsQueryMarLogic` sets the QUERY_TYPE in the test. That test relies on this line in setup. I think it would be better to move this line into the test.

Reply:
Left the set QUERY_TYPE line in the setup method, but add it explicitly to the testCollectionsQueryMarkLogic method. This ensures that at set up a valid processor has a valid configuration and properties can be adjusted as needed in the tests.